src/cast_core/src/actors/CrudActor.cpp

```diff
@@ -0,0 +1,1241 @@
+                              metrics::Operation operation) {
+        if (!opNode.IsMap())
+            return;
+        if (opNode["Options"]) {
+            _options = opNode["Options"].as<mongocxx::options::transaction>();
+        }
+    }
+
+    void run(mongocxx::client_session& session) override {
+        try {
+            session.start_transaction(_options);
+        } catch (mongocxx::operation_exception& exception) {
+            throw;
+        }
+    }
+
+private:
+    mongocxx::options::transaction _options;
+};
+
+/**
+ * Example usage:
+ *    Operations:
+ *    - OperationName: commitTransaction
+ */
+
+struct CommitTransactionOperation : public BaseOperation {
+
+    CommitTransactionOperation(YAML::Node opNode,
+                               bool onSession,
+                               mongocxx::collection collection,
+                               genny::DefaultRandom& rng,
+                               metrics::Operation operation) {}
+
+    void run(mongocxx::client_session& session) override {
+        session.commit_transaction();
+    }
+};
+
+/**
+ * Example usage:
+ *    Operations:
+ *    - OperationName: setReadConcern
+ *      OperationCommand:
+ *        ReadConcern:
+ *          Level: majority
+ */
+
+struct SetReadConcernOperation : public BaseOperation {
+
+    static bool isValidReadConcernString(std::string_view rcString) {
+        return (rcString == "local" || rcString == "majority" || rcString == "linearizable" ||
+                rcString == "snapshot" || rcString == "available");
+    }
+
+    SetReadConcernOperation(YAML::Node opNode,
+                            bool onSession,
+                            mongocxx::collection collection,
+                            genny::DefaultRandom& rng,
+                            metrics::Operation operation)
+        : _collection{collection} {
+        if (!opNode["ReadConcern"]) {
+            throw InvalidConfigurationException(
+                "'setReadConcern' operation expects a 'ReadConcern' field.");
+        }
+        _readConcern = opNode["ReadConcern"].as<mongocxx::read_concern>();
+    }
+
+    void run(mongocxx::client_session& session) override {
+        _collection.read_concern(_readConcern);
+    }
+
+private:
+    mongocxx::collection _collection;
+    mongocxx::read_concern _readConcern;
+};
+
+/**
+ * Example usage:
+ *    Operations:
+ *    - OperationName: drop
+ *      OperationCommand:
+ *        OnSession: false
+ *        Options:
+ *          WriteConcern:
+ *            Level: majority
+ */
+struct DropOperation : public BaseOperation {
+
+    DropOperation(YAML::Node opNode,
+                  bool onSession,
+                  mongocxx::collection collection,
+                  genny::DefaultRandom& rng,
+                  metrics::Operation operation)
+        : _onSession{onSession}, _collection{collection}, _operation{operation} {
+        if (!opNode)
+            return;
```

Comment:
Should this throw an error?

Reply:
No, I decided to make it that for a `collection.drop()` command, the user doesn't have to specify the `OperationCommand` block unless specified with options (like write concern) as the command takes no arguments .